src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -822,11 +822,13 @@ private void orderBy(Compiler compiler) {
     }
 
     private void limit(Compiler compiler) {
-        if (limit > 0 && db.hasCapability(Capability.LIMIT)) {
+        //if a skip value is set but no limit, we have to set a limit > 0 to make limit working.
+        int effectiveLimit = limit == 0 && skip > 0 ? Integer.MAX_VALUE : limit;
```

Comment:
This would always set a limit. 
Refactor it a bit more verbose. Maybe to check first for capability, then for skip (bring in effective limit only there) and then for limit?

Reply:
Code should work. But the ternary with the and condition is maybe hard to read.